server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2241,6 +2241,7 @@ private IndexWriterConfig getIndexWriterConfig() {
             mergePolicy = new ShuffleForcedMergePolicy(mergePolicy);
         }
         iwc.setMergePolicy(mergePolicy);
+        iwc.setMaxFullFlushMergeWaitMillis(-1);
```

Comment:
one quick question, what does `-1` mean here? It says `0` to disable.

Reply:
It's also disabling it. I had grepped the Lucene code and saw some `maxFullFlushMergeWaitMillis > 0` and went with -1 instead of 0.

I'll change it to 0 to stick to the documented value.